File: backend/meks/api/schemas/analytics.py
from pydantic import BaseModel


class AnalyticsQueryRequest(BaseModel):
    query: str
    knowledge_base_ids: list[str] | None = None


class AnalyticsQueryResponse(BaseModel):
    intent_type: str
    columns: list[str]
    rows: list[dict]
    total: int
    query: str
    duration_ms: int
    semantic_results: list | None = None

File: frontend/src/api/analytics.ts
import apiClient from './client'

export interface AnalyticsQuery {
  query: string
  knowledge_base_ids?: string[]
}

export interface AnalyticsResponse {
  columns: string[]
  rows: Record<string, unknown>[]
  intent_type: string
  duration_ms: number
  total: number
}

export const analyticsApi = {
  query: (data: AnalyticsQuery) =>
    apiClient.post<AnalyticsResponse>('/analytics/query', data),
}
